inc/group_group.class.php

```diff
@@ -155,14 +161,24 @@ function getGroups($ticket_id, $removeAlreadyAssigned = true) {
       if (!empty($groups) && count(array_intersect($groups, $user_groups)) == 0) {
          return [];
       }
-
-      //get all group which we can climb
-      if (count($groups) > 0) {
-         $group_group = $this->find(['groups_id_source' => $groups]);
-         foreach ($group_group as $current_group) {
-            $groups[$current_group['groups_id_destination']] = $current_group['groups_id_destination'];
-         }
-      }
+       //get all group which we can climb
+       if ($_SESSION['plugins']['escalade']['config']['use_filter_assign_group']) {
+           if (count($user_groups) > 0) {
+               $group_group = $this->find(['groups_id_source' => $user_groups]);
+               foreach ($group_group as $current_group) {
+                   $groups[$current_group['groups_id_destination']] = $current_group['groups_id_destination'];
+               }
+           }
+       } else {
+           //Retrieve all assignable groups
+           $criteria = [
+               'is_assign' => 1
+           ];
+           $all = getAllDataFromTable(Group::getTable(), $criteria);
+           foreach ($all as $current_group) {
+               $groups[$current_group['id']] = $current_group['id'];
+           }
+       }
```

Comment:
I have two concerns on this:
* It will retrieve all groups regardless of their entities.
* Will this return groups that are already assigned to the ticket ?

I am not very familiar with this plugin behavior but IMO it should replace the groups values only if `use_filter_assign_group` is enabled.
If it is disabled, we keep the original values from GLPI's core (which we shouldn't have to rebuild here).

Do you see what I mean ?

Reply:
If I summarize, we use the native function for retrieving user groups when the filter is not activated, otherwise we retrieve those defined via the group configuration. But in this case, a user can only escalate to their own groups, I'm not sure that this corresponds to the initial functioning of the plugin.

Reply:
Thoughts @stonebuzz ? I am not sure what is the expected behavior here.